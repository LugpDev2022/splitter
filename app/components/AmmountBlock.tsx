import CostSpan from './spans/CostSpan';

type AmmountLabel = 'Tip Ammount' | 'Total';

interface AmmountBlock {
  ammount: number;
  label: AmmountLabel;
}

const AmmountBlock: React.FC<AmmountBlock> = ({ label, ammount }) => (
  <div className='flex flex-col justify-between mb-8'>
    <div className='flex flex-col mb-2'>
      <span className='text-white font-bold tracking-wide'>{label}</span>
      <span className='text-[#7f9c9f]'>/ person</span>
    </div>
    <CostSpan quantity={ammount} />
  </div>
);

export default AmmountBlock;
