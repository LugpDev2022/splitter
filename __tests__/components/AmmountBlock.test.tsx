import AmmountBlock from '@/app/components/AmmountBlock';
import { render, screen } from '@testing-library/react';

describe('Tests on <AmmountBlock />', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <AmmountBlock ammount={0} label='Tip Ammount' />
    );
    expect(container).toMatchSnapshot();
  });

  test('should have the given ammount', () => {
    const ammount = 1057;

    render(<AmmountBlock ammount={ammount} label='Tip Ammount' />);
    expect(screen.getByText('$' + ammount)).toBeTruthy();
  });

  test('should have the given label', () => {
    const label = 'Tip Ammount';

    render(<AmmountBlock ammount={0} label={label} />);
    expect(screen.getByText(label)).toBeTruthy();
  });
});
