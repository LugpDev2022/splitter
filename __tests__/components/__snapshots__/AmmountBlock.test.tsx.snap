// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AmmountBlock /> should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-between mb-8"
  >
    <div
      class="flex flex-col mb-2"
    >
      <span
        class="text-white font-bold tracking-wide"
      >
        Tip Ammount
      </span>
      <span
        class="text-[#7f9c9f]"
      >
        / person
      </span>
    </div>
    <span
      class="text-teal-500 text-5xl font-bold"
    >
      $
      0.00
    </span>
  </div>
</div>
`;
